ita-social-projects/GreenCityUBS
# Geocoding bugfix

dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)


## Code reviewers

- [ ] @github_username

## Summary of issue

When you are adding while ordering a UBS Courier and that address cannot be found, the weird uncorrect address will be added

## Summary of change

1. Deleted method getResultFromGeoCode from GoogleApiService because it didn`t work well, when more than one result is retrieved from geocoding
2. Instead i added two methods that retrieve search results in ukrainian and english.
3. Slightly changed method getLocationDto in UBSClientServiceImpl to be compatible with new changes in GoogleApiService
4. Created method retrieveCorrectDtoRequest in UBSClientServiceImpl that retrieves us address we want among the ones that geocoder returned us or throws AddressNotFoundException if address we search cannot be found
5. Fixed tests

## Testing approach

1. Run GreencityUBS, GreencityUser
2. Login as user
3. Go to save-order-address endpoint
4. Try to put 
{
  "addressComment": "",
  "district": "Солом'янський район",
  "districtEn": "Solom`yans`kyi district",
  "entranceNumber": "2",
  "houseCorpus": "2",
  "houseNumber": "1",
  "region": "Київська область",
  "regionEn": "Kyiv Oblast",
  "searchAddress": "Kyivska St, 1, Kyiv, Ukraine, 02000"
}
You should get AddressNotFoundException
5. Repeat with
{
  "addressComment":"",
  "districtEn":"Holosiivs'kyi district",
  "district":"Голосіївський район",
  "entranceNumber":"8",
  "houseCorpus":"3",
  "houseNumber":"1",
  "regionEn":"Kyiv Oblast",
  "region":"Київська область",
  "searchAddress":"Antonovycha St, 1, Kyiv, Ukraine, 02000"
}
Everything should be fine

## CHECK LIST
- [ ] JIRA/ Github Issue number & title in PR title (ISSUE-#5058: Ticket title)
